src/Robo/Plugin/Commands/DevelopmentModeCommands.php

```diff
@@ -77,12 +116,219 @@ public function devRefreshAll(
             $result = $this->devRefreshDrupal(
                 environmentType: LocalDevEnvironmentTypes::from($environmentType),
                 siteName: $siteName,
-                startLocalEnv: $startLocalDev,
             );
         }
         return $result;
     }
 
+    /**
+     * Refresh a site database in DDEV.
+     *
+     * @param string $siteName
+     *   The Drupal site name.
+     * @option db
+     *   Provide a path to a database dump to be used instead of downloading the latest dump.
+     */
+    public function databaseRefreshDdev(string $siteName = 'default', array $options = ['db' => '']): Result
+    {
+        // @todo: Update this method to not be DDEV specific.
+        $this->io()->title('DDEV database refresh.');
+
+        ['db' => $dbPath] = $options;
+        // Track whether a database path was provided by the user or not.
+        $dbPathProvidedByUser = $dbPath !== '';
+
+        if (!$dbPathProvidedByUser) {
+            $dbPath = $this->databaseDownload($siteName);
+        }
+
+        $this->io()->section("importing $siteName database.");
+        $this->say("Importing $dbPath");
+        $this->taskExec(LocalDevEnvironmentTypes::DDEV->value)
+            ->arg('import-db')
+            ->option('database', $siteName == 'default' ? 'db' : $siteName)
+            ->option('file', $dbPath)
+            ->run();
+
+        // If a database was downloaded as part of this process, delete it.
```

Comment:
When this was originally created, the databases used were small enough that redownloading was rarely an issue. Not deleting makes a lot of sense when we are dealing with very large databases. 

One concession I made towards this was only deleting the database if you are using one of the "wrapper" commands like this one, `database:refresh-ddev` or _its_ wrapper `dev:refresh`. If you call `database:download` directly the file will not be deleted, and then if you call these wrapper commands it will use that file assuming it is the latest database dump in S3.

Reply:
This was a smart decision.